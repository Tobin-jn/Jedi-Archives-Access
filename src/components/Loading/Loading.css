.loading {
  margin: auto;
  grid-column: 1/5;
  align-self: end;
}

.wrapper {
  position: relative;
}

.dot, 
.dot-two {
  position: absolute;
  top: -4.5px;
  left: 50%;
  width: 8px;
  height: 8px;
  border-radius: 50%;
  background: #ADD8E6;
  box-shadow: 0 0 10px .1px #ADD8E6;  
}

.dot-two {
  top: 77px;
}

.loading-wrapper {
  position: relative;
  width: 80px;
  height: 80px;
  
  border: 1px solid #ADD8E6;
  border-radius: 50%;
  box-shadow: 0 0 10px .1px #ADD8E6;  
  
  animation-name: loader;
  animation-duration: 1.5s;
  animation-iteration-count: infinite;
  animation-timing-function: linear;
}

@keyframes loader {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

.icon {
  top: 12px;
  left: 14px;
  position: absolute;
  color: #ADD8E6;
  font-size: 55px;
  animation-name: icon;
  animation-duration: 6s;
  animation-iteration-count: infinite;
  animation-timing-function: linear;
  border-radius: 50%;
  height: 55px;
  width: 55px;
  text-align: center;
  line-height: 1;
}

@keyframes icon {
  0%,
  5%,
  95%,
  100% {
    opacity: 0;
  }
  45%,
  55% {
    opacity: 1;
/*     box-shadow: 0 3px 7px 0 #ADD8E6;   */
  }
}