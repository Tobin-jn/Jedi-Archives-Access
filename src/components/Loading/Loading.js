import React from 'react';
import './Loading.css'

const Loading = () => {
  return(
    <div 
      className="loading"
      aria-label='loading-image'
    >
      <div className='wrapper'>
      <div className='loading-wrapper'>
      <div className="circle"></div>
      <div className="dot"></div>
      <div className="dot-two"></div>
    </div>
    <i className="fab fa-jedi-order icon"></i>
    </div>
    </div>
  )
}

export default Loading