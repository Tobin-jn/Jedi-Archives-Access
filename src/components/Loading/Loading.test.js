import React from 'react';
import Loading from './Loading';
import { shallow } from 'enzyme';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<Loading 
  />)
});

describe('Loading', () => {
  it ('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});