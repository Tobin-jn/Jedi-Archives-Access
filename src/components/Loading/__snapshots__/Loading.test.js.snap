// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should match the snapshot 1`] = `
<div
  aria-label="loading-image"
  className="loading"
>
  <div
    className="wrapper"
  >
    <div
      className="loading-wrapper"
    >
      <div
        className="circle"
      />
      <div
        className="dot"
      />
      <div
        className="dot-two"
      />
    </div>
    <i
      className="fab fa-jedi-order icon"
    />
  </div>
</div>
`;
